documentation/testing/unit-tests.md

```diff
@@ -0,0 +1,194 @@
+        buildParams = emptyBuildParams(),
+        router = router,
+    )
+
+    @AfterEach
+    fun after() {
+        routerTestHelper.close()
+    }
+
+    @Test
+    fun `WHEN resolve Child THEN create Child1 and attach it`() {
+        backStack.replace(Configuration.Content.Child)
+
+        // Node was created and attached to the parent
+        child1Builder.assertLastNodeState(Lifecycle.State.CREATED)
+    }
+
+    @Test
+    fun `WHEN resolve Child THEN create Child1 with proper parameter`() {
+        backStack.replace(Configuration.Content.Child)
+
+        // Node creation was invoked with following parameter
+        child1Builder.assertLastParam(Child1Builder.Param(true))
+    }
+
+}
+```
+
+Most of the time, `Router` is a pretty simple class that does not have any complex logic to test, but only simple mappings from configuration to builder invocation.
+If it is so, then there is no need to cover this class with tests.
+
+## Interactor tests
+
+`Interactor` might have pretty complicated logic related to wiring components like operations on backstack, so it is a great idea to cover it with tests.
+To help you with `Interactor` tests the framework provides `RibInteractorTestHelper`.
+
+The following `Interactor` listens to UI events and child RIB output for routing coordination. 
+
+```kotlin
+class SomeScreenInteractor(
+    buildParams: BuildParams<*>,
+    private val backStack: BackStack<Configuration>
+) : Interactor<SomeScreen, SomeScreenView>(
+    buildParams = buildParams
+) {
+
+    override fun onViewCreated(view: SomeScreenView, viewLifecycle: Lifecycle) {
+        view.subscribe { event ->
+            when (event) {
+                SomeScreenView.Event.OpenChildClicked -> backStack.replace(Configuration.Content.Child1)
+            }
+        }
+    }
+
+    override fun onChildBuilt(child: Node<*>) {
+        when (child) {
+            is Child1 ->
+                child
+                    .output
+                    .subscribe { navigateUp() }
+        }
+    }
+
+}
+```
+
+We can test the logic above with help of the following test.
+
+```kotlin
+class SomeScreenInteractorTest {
+
+    // We can use real BackStack as routing source
+    private val backStack = BackStack<Configuration>(Configuration.Content.Empty, BuildParams.Empty())
+    
+    // Interactor that we test
+    private val interactor = SomeScreenInteractor(
+        buildParams = BuildParams.Empty(),
+        feature = feature,
+        backStack = backStack,
+    )
+
+    // We can replace view implementation with stub, so we can run this test without Robolectric
+    // RibViewStub is suitable for MVI approach, but you can easily create your own View stub just by implementing View interface
+    private val view = object : RibViewStub<SomeScreenView.ViewModel, SomeScreenView.Event>(), SomeScreenView {}
+
+    // RibInteractorTestHelper will invoke all required callbacks to setup Interactor properly
```

Comment:
How about "Utility helper that will allow you to invoke all required callbacks in order to setup the Node's state properly", or something similar?
The way it is written right now makes it sound like it is gonna setup the Node properly by itself.